classes/class-p4-context-controller.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * P4 Context Controller Class
+ *
+ * @package P4MT
+ */
+
+/**
+ * Class P4_Context_Controller Sets common context fields.
+ */
+class P4_Context_Controller {
+
+	/**
+	 * Set context relating to the header
+	 *
+	 * @param array $context To be set.
+	 * @param array $page_meta_data  meta data of page.
+	 */
+	public static function set_header_context( &$context, $page_meta_data ) {
+		$context['header_title']                = is_front_page() ? ( $page_meta_data['p4_title'] ?? '' ) : ( $page_meta_data['p4_title'] ?? $post->title );
+		$context['header_subtitle']             = $page_meta_data['p4_subtitle'] ?? '';
+		$context['header_description']          = wpautop( $page_meta_data['p4_description'] ?? '' );
+		$context['header_button_title']         = $page_meta_data['p4_button_title'] ?? '';
+		$context['header_button_link']          = $page_meta_data['p4_button_link'] ?? '';
+		$context['header_button_link_checkbox'] = $page_meta_data['p4_button_link_checkbox'] ?? '';
+	}
+
+	/**
+	 * Set context fileds relating to the background image.
+	 *
+	 * @param array $context To be set.
+	 */
+	public static function set_background_image_context( &$context ) {
+		$background_image_id                = get_post_meta( get_the_ID(), 'background_image_id', 1 );
+		$context['background_image']        = wp_get_attachment_url( $background_image_id );
+		$context['background_image_srcset'] = wp_get_attachment_image_srcset( $background_image_id, 'full' );
+	}
+
+	/**
+	 * Set open graph context fields.
+	 *
+	 * @param array  $context To be set.
+	 * @param object $post That the context refers to.
+	 */
+	public static function set_og_meta_fields( &$context, $post ) {
+		$context['og_title']       = $post->get_og_title();
+		$context['og_description'] = $post->get_og_description();
+		$context['og_image_data']  = $post->get_og_image();
+	}
+
+	/**
+	 * Set the context fields relating to the data layer.
+	 *
+	 * @param array $context Context to be set.
+	 * @param array $meta Meta data of the page.
+	 */
+	public static function set_campaign_datalayer_context( &$context, $meta ) {
```

Comment:
Maybe you can do like in `set_og_meta_fields` and leave out the final `_context` in this and other functions? There's already `context` in the class name and variable name.